api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCalculatorTest.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.dhsc.htbhf.claimant.service.payments;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PaymentCycleVoucherEntitlement;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PaymentCycleVoucherEntitlementTestDataFactory.aPaymentCycleVoucherEntitlementWithFourWeeklyVouchers;
+
+class PaymentCalculatorTest {
+
+    private int numberOfCalculationPeriods = 4;
+    private PaymentCalculator paymentCalculator = new PaymentCalculator(numberOfCalculationPeriods);
+
+    @Test
+    void shouldReturnFullPaymentAmountForLowBalance() {
+        // Given
+        PaymentCycleVoucherEntitlement entitlement = aPaymentCycleVoucherEntitlementWithFourWeeklyVouchers();
+        int firstWeekVoucherEntitlementAmount = entitlement.getFirstVoucherEntitlementForCycle().getTotalVoucherValueInPence();
+        int lowBalance = 1;
+
+        // When
+        int paymentAmount = paymentCalculator.calculatePaymentAmountCycleInPence(entitlement, lowBalance);
+
+        // Then
+        assertThat(paymentAmount).isEqualTo(firstWeekVoucherEntitlementAmount * 4);
+    }
+
+    @Test
+    void shouldReturnPartialPaymentAmountForBalanceBetweenFourAndEightWeeksEntitlement() {
+        // Given
+        PaymentCycleVoucherEntitlement entitlement = aPaymentCycleVoucherEntitlementWithFourWeeklyVouchers();
```

Comment:
Can we be more explicit about what the total entitlement & max balance amounts are? I'm thinking something along the lines of (for example, not these exact values):
Given currentBalance = 20,000, maxBalance = 30,000, totalVoucherEntitlement = 25,000, paymentAmount = 10,000, or
currentBalance = 29,900, maxBalance = 30,000, totalVoucherEntitlement = 10,000, paymentAmount = 100
This might be a good candidate for a parameterised test, actually.

Reply:
Note that in some of the above cases the totalVoucherAmount isn't 4 * weekly amount - assumption here is that one of the later weeks has a different amount. If that gets really difficult to do in the factory then don't worry about it too much.

Reply:
This is quite difficult to achieve as a lot of the values are calculated internally in each voucher entitlement (which are inside a list property on PaymentCycleVoucherEntitlement). Instead we have refactored the test by adding constants to make the value assertions clearer.